#! /bin/sh
#  # Code and files for the publication "*Factors structuring the epiphytic archaeal and fungal communities in a semi-arid Mediterranean ecosystem*" ^[Katsoula et al. 2021]
#  
#  This repository contains all the necessary code and instructions for reproducing the results found in our study (see citation section). 
#  
#  For running the necessary commands the users need to open a terminal, git-clone or download the repository, and enter the base folder. E.g:
#  
#  ```
#  git clone https://github.com/SotiriosVasileiadis/armenistis.git
#  cd armenistis
#  ```
#  The user, with the "armenistis" folder as working directory, and assumming that the necessary software (described in the next section) is installed, can then run all commands interactively as described here.
#  
#  The users also need to find the associated databases as described in the manuscript and need to install all R packages found in the **library()** load commands in the provided R scripts at the statistical analysis part.
#  
#  ## Software required for running the various bash script modules
#  
#  The following software is necessary for running all scripts along with R packages found in the scripts in th 5_statistical_analysis sub-folders
#  
#  [perl v5.32](https://metacpan.org/release/perl)
#  
#  [Edirect tools v14.2](https://dataguide.nlm.nih.gov/edirect/install.html)
#  
#  [SRA toolkit v2.10.9](https://github.com/ncbi/sra-tools/wiki/02.-Installing-SRA-Toolkit)
#  
#  [Flexbar v3.5](https://github.com/seqan/flexbar)
#  
#  Piping own multiplexing system demultiplexing Script created by co-author Sotirios Vasileiadis DemuxOwnBCsys_absPATH.sh (provided in the supplied material)
#  
#  [TrimmomaticPE v0.39](http://www.usadellab.org/cms/?page=trimmomatic)
#  
#  [FLASH v1.2.11](https://ccb.jhu.edu/software/FLASH/#:~:text=FLASH%20is%20designed%20to%20merge,to%20merge%20RNA-seq%20data.)
#  
#  [LotuS v1.58](http://psbweb05.psb.ugent.be/lotus/)
#  
#  
#  ### 1) Download and merge data in order to mimic the analysis performed in the mansucript
#  
#  
#  Install the Edirect tools v14.2 according to the instructions of https://dataguide.nlm.nih.gov/edirect/install.html
#  
#  Enter the 1_download_data folder
#  ```
cd 1_download_data
#  ```
#  
#  Download the bioproject run accessions and create a list
#  ```
esearch -db sra -query PRJNA531404 | \
efetch -format runinfo | \
cut -d "," -f 1 | grep SRR > SRR_accessions.txt
#  ```
#  
#  Install the SRA toolkit v2.10.9 according to the instructions of https://github.com/ncbi/sra-tools/wiki/02.-Installing-SRA-Toolkit
#  
#  Then, download the fastq in split forward/reverse format
#  ```
for i in `cat SRR_accessions.txt`
do
	fastq-dump -I --split-files ${i}
done
#  ```
#  Concatenate the files to a single library parted by a generic forward reads file and a generic reverse reads file to mimic the files generated by the sequencing center and store them in the 2_demultiplex_data folder
#  ```
cat *_1.fastq > ../2_demultiplex_data/R1.fastq
cat *_2.fastq > ../2_demultiplex_data/R2.fastq

cd ..
#  ```
#  ### 2) Demultiplex data using the script prepared by Sotirios Vasileiadis which is based on the [Flexbar v3.5 software](https://github.com/seqan/flexbar)
#  Start with the prokaryotes
#  ```
cd 2_demultiplex_data/prok
#  ```
#  You can test the "DemuxOwnBCsys_absPATH.sh" for necessary arguments with "sh DemuxOwnBCsys_absPATH.sh"
#  Run the demultiplexing process by calling the "run_demux.sh" bash script which is created by the run_demux_pre.sh script after replacing the absolute path value with the current directory path 
#  ```
perl -pe "s|absolutepath|\\Q${PWD}|g" run_demux_pre.sh | perl -pe "s/\\\//g" > run_demux.sh
#  ```
#  Attention!!! the generated last two files (last index read pairs) will remain gnu-zipped. This is a minor pipeline glitch and you need to unzip them manually or add a script.
#  ```
sh run_demux.sh

cd ../../
#  ```
#  Then delmultiplex the sequences for the fungal dataset
#  ```
cd 2_demultiplex_data/fung
#  ```
#  You can test the "DemuxOwnBCsys_absPATH.sh" for necessary arguments with "sh DemuxOwnBCsys_absPATH.sh"
#  Run the demultiplexing process by calling the "run_demux.sh" bash script which is created by the run_demux_pre.sh script after replacing the absolute path value with the current directory path 
#  ```
perl -pe "s|absolutepath|\\Q${PWD}|g" run_demux_pre.sh | perl -pe "s/\\\//g" > run_demux.sh
sh run_demux.sh
#  ```
#  Attention!!! the generated last two files (last index read pairs) will remain gnu-zipped. This is a minor pipeline glitch and you need to unzip them manually or add a loop
#  ```
cd ../../
#  ```
#  
#  ### 3) Quality trim the sequence reads with [TrimmomaticPE v0.39](http://www.usadellab.org/cms/?page=trimmomatic) and pair assemble with [FLASH v1.2.11](https://ccb.jhu.edu/software/FLASH/#:~:text=FLASH%20is%20designed%20to%20merge,to%20merge%20RNA-seq%20data.)
#  ```
cd 3_qc_trim_pair_assemble/prok

for i in `cat mylist`
	do
	TrimmomaticPE -threads 4 -phred33 ../../2_demultiplex_data/prok/NP2_out/lOTUs_moth_basis/${i}_lib_R1.fastq ../../2_demultiplex_data/prok/NP2_out/lOTUs_moth_basis/${i}_lib_R2.fastq trimmed/${i}_tr_R1.fastq trimmed/${i}_tr_R1_unpaired.fastq trimmed/${i}_tr_R2.fastq trimmed/${i}_tr_R2_unpaired.fastq LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:100
	flash -m 10 -M 250 -x 0.1 -p 33 -d combined_merged -o ${i} --compress-prog=gzip --suffix=gz -t 4 trimmed/${i}_tr_R1.fastq trimmed/${i}_tr_R2.fastq
	done

cd 3_qc_trim_pair_assemble/fung

for i in `cat mylist`
	do
	TrimmomaticPE -threads 4 -phred33 ../../2_demultiplex_data/fung/NP2_out/lOTUs_moth_basis/${i}_lib_R1.fastq ../../2_demultiplex_data/fung/NP2_out/lOTUs_moth_basis/${i}_lib_R2.fastq trimmed/${i}_tr_R1.fastq trimmed/${i}_tr_R1_unpaired.fastq trimmed/${i}_tr_R2.fastq trimmed/${i}_tr_R2_unpaired.fastq LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:100
	flash -m 10 -M 250 -x 0.1 -p 33 -d combined_merged -o ${i} --compress-prog=gzip --suffix=gz -t 4 trimmed/${i}_tr_R1.fastq trimmed/${i}_tr_R2.fastq
	done
#  ```
#  
#  ### 4) Prep of the OTU and taxonomy matrices with [LotuS v1.58](http://psbweb05.psb.ugent.be/lotus/) wrapper
#  ```
cd 4_run_lotus/prok
lotus.pl -i ../../3_qc_trim_pair_assemble/prok/combined_merged -o lotus_out -m map -p hiSeq -c lOTUs.cfg -s sdm_hiSeq.txt -t lotus_tmp -threads 8 -refDB SLV -tax_group bacteria -amplicon_type SSU -useBestBlastHitOnly 0 -simBasedTaxo 2 -keepUnclassified 1 -LCA_frac 0.3
cd ../../

cd 4_run_lotus/fung
lotus.pl -i ../../3_qc_trim_pair_assemble/fung/combined_merged -o lOTUs_UNITE_8.2 -m map -p hiSeq -c lOTUs_UNITE_8.2.cfg -s sdm_hiSeq.txt -t lotus_tmp -threads 8 -refDB UNITE -tax_group fungi -amplicon_type ITS -useBestBlastHitOnly 0 -simBasedTaxo 2 -keepUnclassified 1 -LCA_frac 0.3
cd ../../
#  ```
#  
#  ### 5) Run the statistical analysis
#  
#  For the performance of this part you need to obtain the OTU.txt and hiera_BLAST.txt files from each LotuS output (or use those originally generated by us and provided in the corresponding folders), place them in the corresponding sub-folders of the "5_statistical_analysis", and run the scripts.R file for [archea](5_statistical_analysis/arch/scripts.R) and [fungi](5_statistical_analysis/fung/scripts.R), or the [ANOVA_qpcr.R](5_statistical_analysis/qpcr_results_stats/ANOVA_qpcr.R) file. As mentioned also previously, it is necessary to install all packages declared in the "library()" package-calling commands throughout the scripts.
#  
#  # Citation
#  
#  Katsoula A, Vasileiadis S, Karamanoli K, Vokou D\*, Karpouzas DG\* (2021) Factors structuring the epiphytic archaeal and fungal communities in a semi-arid Mediterranean ecosystem. Microbial Ecology
#  
#  \* corresponding authors 
#  vokou@bio.auth.gr, dkarpouzas@uth.gr
#  
#  # Code usage disclaimer 
#  
#  The following is the disclaimer that applies to all scripts, functions, one-liners, etc. This disclaimer supersedes any disclaimer included in any script, function, one-liner, etc.
#  
#  You running this script/function means you will not blame the author(s) if this breaks your stuff. This script/function is provided **AS IS** without warranty of any kind. Author(s) disclaim all implied warranties including, without limitation, any implied warranties of merchantability or of fitness for a particular purpose. The entire risk arising out of the use or performance of the sample scripts and documentation remains with you. In no event shall author(s) be held liable for any damages whatsoever (including, without limitation, damages for loss of business profits, business interruption, loss of business information, or other pecuniary loss) arising out of the use of or inability to use the script or documentation. Neither this script/function, nor any part of it other than those parts that are explicitly copied from others, may be republished without author(s) express written permission. Author(s) retain the right to alter this disclaimer at any time. For the most up to date version of the disclaimer, see https://ucunleashed.com/code-disclaimer.
#  
